reaktive/src/commonTest/kotlin/com/badoo/reaktive/observable/TakeObservableTests.kt

```diff
@@ -0,0 +1,109 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.test.base.assertDisposed
+import com.badoo.reaktive.test.base.assertError
+import com.badoo.reaktive.test.base.assertNotError
+import com.badoo.reaktive.test.observable.*
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+class TakeObservableTests :
+    ObservableToObservableTests by ObservableToObservableTests<Unit>({ take(1) }) {
+
+    private val upstream = TestObservable<Int?>()
+    val observer = upstream.take(10).test()
+
+    @Test
+    fun completes_WHEN_limit_of_1_is_reached() {
+        val observer = upstream.take(1).test()
+
+        upstream.onNext(42)
+
+        observer.assertComplete()
+        observer.assertNotError()
+    }
+
+    @Test
+    fun completes_WHEN_limit_of_2_is_reached() {
+        val observer = upstream.take(2).test()
+
+        upstream.onNext(42)
+        observer.assertNotComplete()
+
+        upstream.onNext(84)
+        observer.assertComplete()
+        observer.assertNotError()
+    }
+
+    @Test
+    fun completes_WHEN_upstream_completed() {
+        upstream.onComplete()
+
+        observer.assertComplete()
+    }
+
+    @Test
+    fun produces_same_error_WHEN_upstream_produced_error() {
+        val error = Exception()
+
+        upstream.onError(error)
+
+        observer.assertError(error)
+    }
+
+    @Test
+    fun disposes_WHEN_upstream_disposed() {
+        upstream.dispose()
+
+        observer.assertDisposed()
+    }
+
+    @Test
+    fun produces_values_in_correct_order_from_upstream() {
+        upstream.onNext(42, null, 84, null)
+
+        observer.assertValues(42, null, 84, null)
+    }
+
+    @Test
+    fun multiple_takes() {
+        val observer = upstream
+            .take(100)
+            .take(1)
+            .test()
+
+        (1..100).forEach {
+            upstream.onNext(it)
+        }
+
+        observer.assertValue(1)
+        observer.assertComplete()
+    }
+
+    @Test
+    fun re_entrant_take() {
+        val observer = upstream
+            .take(1)
+            .doOnBeforeNext {
+                upstream.onNext(84)
+            }
+            .test()
+
+        upstream.onNext(42)
+
+        observer.assertValue(42)
+        observer.assertComplete()
+    }
+
+    @Test fun throws_error_when_take_is_called_with_a_limit_of_0() {
+        assertFailsWith<IllegalArgumentException>("count >= 0 required but it was 0") {
```

Comment:
`count > 0`

And I am not sure about checking exception message, just `IllegalArgumentException` should be fine.

Reply:
0 should be a valid value. Please test for failure on `-1`.

Reply:
Updated the error message. Continuing this conversation here: https://github.com/badoo/Reaktive/pull/193#discussion_r324485860